CONTRIBUTING.md

```diff
@@ -48,6 +48,58 @@ python run-clang-format.py \
 
 where `$CLANG_FORMAT` denotes the path to the downloaded binary.
 
+## Adding Third Party Libraries
+
+The following steps outline how to add third party libraries to torchtext:
+
+1. Add the git repo you care about as a submodule. Here is a great
+   [tutorial](https://www.atlassian.com/git/tutorials/git-submodule) on working with submodules in git.
+   - Navigate to `third_party/` folder and run `git submodule add <repo-URL>`
+   - Verify the newly added module is present in the
+     [`.gitmodules`](https://github.com/pytorch/text/blob/main/.gitmodules) file
+2. Update
+   [`third_party/CMakeLists.txt`](https://github.com/pytorch/text/blob/70fc1040ee40faf129604557107cc59fd51c4fe2/third_party/CMakeLists.txt#L8)
+   to add the following line: `add_subdirectory(<name-of-submodule-folder> EXCLUDE_FROM_ALL)`
+3. (Optional) If any of the files within the `csrc/` folder make use of the newly added third party library then
+   - Add the new submodule folder to
+     [`​​LIBTORCHTEXT_INCLUDE_DIRS`](https://github.com/pytorch/text/blob/70fc1040ee40faf129604557107cc59fd51c4fe2/torchtext/csrc/CMakeLists.txt#L24)
+     and to
+     [`EXTENSION_INCLUDE_DIRS`](https://github.com/pytorch/text/blob/70fc1040ee40faf129604557107cc59fd51c4fe2/torchtext/csrc/CMakeLists.txt#L119)
+   - Add the submodule name to
+     [`LIBTORCHTEXT_LINK_LIBRARIES`](https://github.com/pytorch/text/blob/70fc1040ee40faf129604557107cc59fd51c4fe2/torchtext/csrc/CMakeLists.txt#L33)
+4. Verify the torchtext build works by running `python setup.py develop`
+
+## Adding a Custom C++ Operator
+
+Custom C++ operators can be implemented and registered in torchtext for several reasons including to make an existing
+Python component more efficient, and to get around the limitations when working with multithreading in Python (due to
+the Global Interpreter Lock). These custom kernels (or “ops”) can be embedded into a TorchScripted model and can be
+executed both in Python and in their serialized form directly in C++. You can learn more in this
+[tutorial on writing custom C++ operators](https://pytorch.org/tutorials/advanced/torch_script_custom_ops.html)
+
+Steps to register an operator:
+
+1. Add the new custom operator to the [`torchtext/csrc`](https://github.com/pytorch/text/tree/main/torchtext/csrc)
+   folder. This entails writing the header and the source file for the custom op.
+2. Add the new source files to the
+   [`LIBTORCHTEXT_SOURCES`](https://github.com/pytorch/text/blob/70fc1040ee40faf129604557107cc59fd51c4fe2/torchtext/csrc/CMakeLists.txt#L11)
+   list.
+3. Register the operators with torchbind and pybind
+   - Torchbind registration happens in the
+     [`register_torchbindings.cpp`](https://github.com/pytorch/text/blob/70fc1040ee40faf129604557107cc59fd51c4fe2/torchtext/csrc/register_torchbindings.cpp#L14)
```

Comment:
off-topic: Note that this style of registration is not necessarily the recommended way.
Par-file registration with `TORCH_LIBRARY_FRAGMENT` macro is more flexible.

Reply:
Gotcha. Is there a reason that domain libraries are currently using this type of registration as opposed to using `TORCH_LIBRARY_FRAGMENT`?

Reply:
I think TorchText is left behind on this one.
TorchAudio and TorchVision have moved to `TORCH_LIBRARY_FRAGMENT` a long time ago.